sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileCatalog.scala

```diff
@@ -294,7 +308,7 @@ object PartitioningAwareFileCatalog extends Logging {
   private def listLeafFilesInParallel(
       paths: Seq[Path],
       hadoopConf: Configuration,
-      sparkSession: SparkSession): Seq[FileStatus] = {
+      sparkSession: SparkSession): Map[Path, Seq[FileStatus]] = {
```

Comment:
I believe this should return the same type as `listLeafFilesInSerial`, i.e. `Seq[(Path, Seq[FileStatus])]`.